src/osd/PG.cc

```diff
@@ -2395,6 +2395,15 @@ std::pair<ghobject_t, bool> PG::do_delete_work(
     ++num;
   }
   bool running = true;
+
+  std::lock_guard l(osd->pg_delete);
```

Comment:
how frequent are the calls to collect_pg_stats()?
(i.e. - is there a cheaper place to execute this code?)

Reply:
default 5s, controlled by option `mgr_stats_period`

Reply:
@rzarzynski - can you review (esp this) as well? it might be negligible enough, or we may want to use something more complex here.

Reply:
@ronen-fr I will encapsulate the logic here into two functions, reduced the granularity of pg_delete lock, and the code is also more concise.